starknet_py/tests/e2e/integration_tests/client_test.py

```diff
@@ -0,0 +1,186 @@
+        entry_point_selector="0x02d757788a8d8d6f21d1cd40bce38a8222d70654214e96ff95d8086e684fbee5",
+        payload=[
+            "0x54d01e5fc6eb4e919ceaab6ab6af192e89d1beb4f29d916768c61a4d48e6c95",
+            "0x38d7ea4c68000",
+            "0x0",
+        ],
+    )
+
+    assert isinstance(estimated_message, EstimatedFee)
+
+
+@pytest.mark.asyncio
+async def test_estimate_message_fee_invalid_eth_address_assertion_error(
+    full_node_client_integration,
+):
+    client = full_node_client_integration
+    invalid_eth_address = "0xD"
+
+    # info about this message from
+    # https://integration.starkscan.co/message/0x140185c79e5a04c7c3fae513001f358beb66653dcee75be38f05bd30adba85dd
+    with pytest.raises(
+        AssertionError,
+        match=f"Argument 'from_address': {invalid_eth_address} is not a valid Ethereum address.",
+    ):
+        _ = await client.estimate_message_fee(
+            from_address=invalid_eth_address,
+            block_number=306687,
+            to_address="0x073314940630fd6dcda0d772d4c972c4e0a9946bef9dabf4ef84eda8ef542b82",
+            entry_point_selector="0x02d757788a8d8d6f21d1cd40bce38a8222d70654214e96ff95d8086e684fbee5",
+            payload=[
+                "0x54d01e5fc6eb4e919ceaab6ab6af192e89d1beb4f29d916768c61a4d48e6c95",
+                "0x38d7ea4c68000",
+                "0x0",
+            ],
+        )
+
+
+@pytest.mark.parametrize(
+    "from_address, to_address",
+    (
+        (
+            "0xbe1259ff905cadbbaa62514388b71bdefb8aacc1",
+            "0x2137",
+        ),  # valid `from_address`, invalid `to_address`
+        (
+            "0xDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD",
+            "0x073314940630fd6dcda0d772d4c972c4e0a9946bef9dabf4ef84eda8ef542b82",
+        ),  # invalid `from_address` (passes through assert), valid `to_address`
+    ),
+)
+@pytest.mark.asyncio
+async def test_estimate_message_fee_throws(
+    full_node_client_integration, from_address, to_address
+):
+    client = full_node_client_integration
+
+    with pytest.raises(ClientError):
+        _ = await client.estimate_message_fee(
+            block_number=123123,
+            from_address=from_address,
+            to_address=to_address,
+            entry_point_selector="0x3248",
+            payload=[
+                "0x4359",
+            ],
+        )
+
+
+@pytest.mark.asyncio
+async def test_get_tx_receipt_reverted(client_integration):
+    client = client_integration
+    reverted_tx_hash = (
+        "0x4a3c389816f8544d05db964957eb41a9e3f8c660e8487695cb75438ef983181"
+    )
+
+    res = await client.get_transaction_receipt(tx_hash=reverted_tx_hash)
+
+    assert res.execution_status == TransactionExecutionStatus.REVERTED
+    assert res.finality_status == TransactionFinalityStatus.ACCEPTED_ON_L1
+    if isinstance(client, GatewayClient):
+        assert "Input too long for arguments" in res.revert_error
+    else:
+        assert "Input too long for arguments" in res.rejection_reason
```

Comment:
Same here, shouldn't these be consistent?

Reply:
Nope, those are named differently in gateway and full node💀. However, `rejection_reason` is "our name" - our schema validates field named `revert_reason` and renames it into `rejection_reason` (for no reason).

Reply:
https://github.com/software-mansion/starknet.py/pull/1125/files/4caaea90172e6753a114a64a127ca2998e75adeb#r1287066888